homeassistant/components/voice_assistant/websocket_api.py

```diff
@@ -77,7 +126,15 @@ async def websocket_run(
     # Cancel pipeline if user unsubscribes
     connection.subscriptions[msg["id"]] = run_task.cancel
 
+    # Confirm subscription
     connection.send_result(msg["id"])
 
-    # Task contains a timeout
-    await run_task
+    try:
+        # Task contains a timeout
+        await run_task
+    except PipelineError as err:
+        connection.send_error(msg["id"], err.code, err.message)
```

Comment:
This doesn't do anything, as you already did a `send_result`. This needs to be an event.